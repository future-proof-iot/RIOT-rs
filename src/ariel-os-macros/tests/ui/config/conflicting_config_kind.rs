// #![no_std]
#![no_main]

// This will not get used because the attribute macro is expected to fail
#[allow(unused_imports)]
use ariel_os::reexports::embassy_net;

// FAIL: network and usb cannot be used at the same time
#[ariel_os::config(network, usb)]
const NETWORK_CONFIG: embassy_net::Config = {
    use embassy_net::Ipv4Address;

    embassy_net::Config::ipv4_static(embassy_net::StaticConfigV4 {
        address: embassy_net::Ipv4Cidr::new(Ipv4Address::new(10, 42, 0, 61), 24),
        dns_servers: heapless::Vec::new(),
        gateway: Some(Ipv4Address::new(10, 42, 0, 1)),
    })
};
