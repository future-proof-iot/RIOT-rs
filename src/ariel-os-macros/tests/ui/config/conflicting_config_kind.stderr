error: custom attribute panicked
 --> tests/ui/config/conflicting_config_kind.rs:9:1
  |
9 | #[ariel_os::config(network, usb)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: only one configuration is supported at a time, use a separate constant for `usb` configuration
